charms/istio-pilot/src/charm.py

```diff
@@ -107,16 +106,22 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self.reconcile)
 
         # Watch:
-        # * relation_joined: because we send data to the other side whenever we see a related app
+        # * relation_created: because we send data to the other side whenever we see a related app
+        # * relation_joined: because we send data to the other side whenever we see a related unit
         self.framework.observe(
             self.on[GATEWAY_INFO_RELATION_NAME].relation_created, self.reconcile
         )
+        self.framework.observe(self.on[GATEWAY_INFO_RELATION_NAME].relation_joined, self.reconcile)
+        # Configure the gateway-info provider
+        self.gateway_provider = GatewayProvider(self)
 
         # Watch:
-        # * relation_joined: because we send data to the other side whenever we see a related app
+        # * relation_created: because we send data to the other side whenever we see a related app
+        # * relation_joined: because we send data to the other side whenever we see a related unit
         # * relation_changed: because of SDI's data versioning model, which first agrees on the
         #                     schema version and then sends the rest of the data
         self.framework.observe(self.on["istio-pilot"].relation_created, self.reconcile)
+        self.framework.observe(self.on["istio-pilot"].relation_joined, self.reconcile)
         self.framework.observe(self.on["istio-pilot"].relation_changed, self.reconcile)
 
```

Comment:
Do we want to do anything on `relation-depated`, `relation-broken`?

Reply:
the `istio-pilot` relation is just an outward sending of information, the istio-pilot charm doesn't actually do anything based on who is related.  So I think we can safely ignore `relation-departed` and `relation-broken`?